render/detailed_node.go

```diff
@@ -64,23 +64,40 @@ func (r sortableRows) Less(i, j int) bool {
 	}
 }
 
-type tables []Table
+type sortableTables []Table
 
-func (t tables) Len() int           { return len(t) }
-func (t tables) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
-func (t tables) Less(i, j int) bool { return t[i].Rank > t[j].Rank }
+func (t sortableTables) Len() int           { return len(t) }
+func (t sortableTables) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
+func (t sortableTables) Less(i, j int) bool { return t[i].Rank > t[j].Rank }
 
 // MakeDetailedNode transforms a renderable node to a detailed node. It uses
 // aggregate metadata, plus the set of origin node IDs, to produce tables.
 func MakeDetailedNode(r report.Report, n RenderableNode) DetailedNode {
-	tables := tables{}
+	tables := sortableTables{}
+
+	// Figure out if multiple hosts/containers are referenced by the renderableNode
+	originHosts := make(map[string]struct{})
+	originContainers := make(map[string]struct{})
+	for _, id := range n.Origins {
+		for _, topology := range r.Topologies() {
+			if nmd, ok := topology.NodeMetadatas[id]; ok {
+				originHosts[report.ExtractHostID(nmd)] = struct{}{}
+				if id, ok := nmd.Metadata[docker.ContainerID]; ok {
+					originContainers[id] = struct{}{}
+				}
+			}
+		}
+	}
+	multiHost := len(originHosts) > 1
+	multiContainer := len(originContainers) > 1
```

Comment:
You could move this check into the loop and break out early.

I'd also put this logic in a separate function.